subsys/spm/secure_services.c

```diff
@@ -67,6 +70,13 @@ int spm_request_read(void *destination, u32_t addr, size_t len)
 		/* Allow reads of mcuboot metadata */
 		{.start = PM_MCUBOOT_PAD_ADDRESS,
 		 .size = PM_MCUBOOT_PAD_SIZE},
+#endif
+#if defined(PM_S0_ADDRESS) && defined(PM_S1_ADDRESS)
+		/* Allow read of image metadata used for checking active slot */
+		{.start = PM_S0_IMAGE_ADDRESS + CONFIG_FW_FIRMWARE_INFO_OFFSET,
```

Comment:
3 different offsets are possible and the configuration of SPM should not depend on what configuration of `CONFIG_FW_FIRMWARE_INFO_OFFSET` you chose to compile your previous application/bootloader with. AFIK the `immutable bootloader` checks all 3 offsets before it boots the next step.